### ISSUE TITLE
self.lr_scheduler.step(epoch=epoch-1)and NaNs in training loss after few iterations

### ISSUE THREAD

**Issue by saikiran6** [yassouali/pytorch_segmentation#74]
Hi, trying to understand how the self.lr_scheduler.step(epoch=epoch-1) in inner for loop of _train_peoch function (in trainer.py) helps. what if it is called outside the inner for loop. Also why epoch=epoch-1 is assigned.

My training loss is becoming NaN after few iterations. Tried changing the lr, momentum, SGD to Adam, Poly to OneCycle. But still getting NaNs in loss. 

Iam trying to train the network on 4 gpus.

**Comment by yassouali** [comment]
Hi, the `lr_scheduler` updates the LR per iteration and not per epoch (at each call, `self.cur_iter` is incremented), this is why we call it at each iteration. We set `epoch-1` because the epoch starts from 1 and not 0 (for printing reasons).

As for the NaN loss, can you provide some information, what model / dataset ...
